controllers/jobs.js

```diff
@@ -0,0 +1,77 @@
+const Tires = require('../models/Job')
+const { StatusCodes } = require('http-status-codes')
+const { BadRequestError, NotFoundError } = require('../errors')
+
+
+const getAllJobs = async (req, res) => {
+    const tires = await Tires.find({ createdBy: req.user.userId }).sort('createdAt')
+    res.status(StatusCodes.OK).json({ tires, count: tires.length })
+}
+
+const getJob = async (req, res) => {
+    // searches for the userID and tireID to make sure they match
+    const { user: { userId }, params: { id: tireID }, } = req
+
+    // will locate the tireID that was created by the user searching
+    const tires = await Tires.findOne({ _id: tireID, createdBy: userId })
+
+    if (!tires) {
+        throw new NotFoundError(`No tire found with ${tireID}`)
+    }
+    res.status(StatusCodes.OK).json({ tires })
+}
+
+const createJob = async (req, res) => {
+    req.body.createdBy = req.user.userId
+    const tires = await Tires.create(req.body)
+    res.status(StatusCodes.CREATED).json({ tires })
+}
+
+const updateJob = async (req, res) => {
+    // searches for the userID and tireID to make sure they match
+    const {
+        body: { brand, size, location, price, quantity },
+        user: { userId },
+        params: { id: tireID },
+    } = req
+
+    // will check for blank fields
+    if (brand === "" || size === "" || location === "" || price === "" || quantity === "") {
```

Comment:
Technically validation should take care of this, but not a bad idea to double check. Especially in the update method